src/aof.c

```diff
@@ -2456,6 +2458,15 @@ int rewriteAppendOnlyFileRio(rio *aof) {
             
             /* Get the expire time */
             expiretime = kvobjGetExpire(o);
+
+            /* Skip keys that are being trimmed */
+            if (server.cluster_enabled) {
+                int curr_slot = kvstoreIteratorGetCurrentDictIndex(kvs_it);
+                if (isSlotInTrimJob(curr_slot)) {
+                    skipped++;
+                    continue;
+                }
```

Comment:
perhaps we can skip whole dict

Reply:
yes, for `KEYS` command, we can have similar optimization, but it looks like not critical code path, we can revisit later